plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -107,6 +109,9 @@ async fn run(tedge_config: TEdgeConfig) -> Result<(), anyhow::Error> {
     // Instantiate config manager actor
     let config_manager_config =
         ConfigManagerConfig::from_tedge_config(DEFAULT_TEDGE_CONFIG_PATH, &tedge_config)?;
+
+    validate_config_dir_and_file(config_manager_config.plugin_config_path.as_path())?;
```

Comment:
My feeling is that checking the existence of the config dir would be simpler and more natural if done inside `ConfigManagerConfig::from_tedge_config()` because there you will have the path to the conf directory and not to the configuration file.